contrib/platform/src/com/sun/jna/platform/mac/CoreFoundation.java

```diff
@@ -0,0 +1,573 @@
+     *            buffer contains the converted characters. If there is an error in
+     *            conversion, the buffer contains only partial results.
+     *            <p>
+     *            The buffer must be large enough to contain the converted
+     *            characters and a NUL terminator.
+     * @param bufferSize
+     *            The length of {@code buffer} in bytes.
+     * @param encoding
+     *            The string encoding to which the character contents of
+     *            {@code theString} should be converted. The encoding must specify
+     *            an 8-bit encoding.
+     * @return {@code true} upon success or {@code false} if the conversion fails or
+     *         the provided buffer is too small.
+     */
+    boolean CFStringGetCString(CFTypeRef theString, Pointer bufferToFill, long bufferSize, int encoding);
+
+    /**
+     * Returns the value of a {@code CFBoolean} object.
+     *
+     * @param bool
+     *            The boolean to examine.
+     * @return The value of {@code bool}.
+     */
+    boolean CFBooleanGetValue(CFTypeRef bool);
+
+    /**
+     * Retrieves a value at a given index.
+     *
+     * @param theArray
+     *            The array to examine.
+     * @param idx
+     *            The index of the value to retrieve. If the index is outside the
+     *            index space of {@code theArray} (0 to N-1 inclusive (where N is
+     *            the count of {@code theArray})), the behavior is undefined.
+     * @return The value at the {@code idx} index in {@code theArray}).
+     */
+    CFTypeRef CFArrayGetValueAtIndex(CFArrayRef theArray, int idx);
+
+    /**
+     * Returns the type used by a {@code CFNumber} object to store its value.
+     *
+     * @param number
+     *            The {@code CFNumber} object to examine.
+     * @return A constant that indicates the data type of the value contained in
+     *         number. See {@link CFNumberType} for a list of possible values.
+     */
+    int CFNumberGetType(CFTypeRef number);
+
+    /**
+     * Obtains the value of a {@code CFNumber} object cast to a specified type.
+     *
+     * @param number
+     *            The {@code CFNumber} object to examine.
+     * @param theType
+     *            A constant that specifies the data type to return. See
+     *            {@link CFNumberType} for a list of possible values.
+     * @param On
+     *            return, contains the value of {@code number}.
+     */
+    void CFNumberGetValue(CFTypeRef number, int theType, ByReference valuePtr);
+
+    /**
+     * Returns the number (in terms of UTF-16 code pairs) of Unicode characters in a
+     * string.
+     *
+     * @param theString
+     *            The string to examine.
+     * @return The number (in terms of UTF-16 code pairs) of characters stored in
+     *         {@code theString}.
+     */
+    long CFStringGetLength(CFTypeRef theString);
```

Comment:
According to https://developer.apple.com/documentation/corefoundation/1542853-cfstringgetlength?language=objc the return is CFIndex, in other places a CFIndex was an int. Judging from https://developer.apple.com/documentation/corefoundation/cfindex?language=objc and https://developer.apple.com/library/archive/documentation/Darwin/Conceptual/64bitPorting/transition/transition.html this needs to be a NativeLong (the size varys with platform).

Reply:
Yep, it is/was a `NativeLong` but OSX has been 64-bit since 10.7.  In [previous PRs that I have submitted here](https://github.com/java-native-access/jna/pull/995#issuecomment-412591274) it was suggested that `NativeLong` was unnecessary and `long` was fine.

I did originally map all these as `int` and tried to switch them all to `long` but apparently missed some.

Reply:
From a quick glance: `CFNumberGetValue`s second parameter (`theType`) is mapped as int, according to https://developer.apple.com/documentation/corefoundation/1543114-cfnumbergetvalue it is a `CFNumberType`, which is a typedef  for `CFIndex` (https://developer.apple.com/documentation/corefoundation/cfnumbertype) which maps to a native `long` (https://developer.apple.com/documentation/corefoundation/cfindex?language=objc).

Reply:
Gotcha.  I guess that should properly be `long` although its value comes from an enum that would fit in a `byte`.  I'll fix that.